### ISSUE TITLE
Add tests for testing block_given? functionality of DataFrame.from_csv() method.

### ISSUE THREAD

**Issue by v0dro** [SciRuby/daru#308]
The `DataFrame.from_csv` method currently has a provision for accepting blocks and performing some manipulation on a row that has been read before loading the data into a dataframe.

However, there are no tests in `io_spec.rb` for testing this.

Tests should amply test error conditions too.

**Comment by zverok** [comment]
@v0dro is there a real life reason for this functionality, in the first place?

**Comment by v0dro** [comment]
I recently saw a PR that attempted to remove the line containing `block_given?` and the specs passed since there was no test of this sort.

**Comment by zverok** [comment]
Yes, I understand why we need the test, if functionality exists.
But my question is WHY the functionality exists?

**Comment by v0dro** [comment]
Say you have a CSV column that contains dates in the form `DD MONTH YEAR` (example `12 february 2016`) and you want to convert this to a `DateTime` when you read the file to a dataframe by using your own conversion logic. The easiest way to do it would be by passing a block to `from_csv` that can modify the data as it comes.

**Comment by gnilrets** [comment]
I process a lot of CSV files and have gotten into the habbit of reading all fields as strings and doing conversions after it's built into a dataframe.

**Comment by zverok** [comment]
Got it, thanks :+1:

**Comment by v0dro** [comment]
@gnilrets for smaller dataframes and simpler usage scenarios I think passing a block is more readable and straightforward.

**Comment by GusAndrianos** [comment]
I want to work on this. I'm looking for something I can do for GSoC and I think it's a good fit.

**Comment by v0dro** [comment]
@GusAndrianos yes this would be a great and simple issue to start with. Have you had a look at the source code yet? You should hurry up with your proposal since the deadline for submitting the final proposal is 4th April.

**Comment by GusAndrianos** [comment]
@v0dro This wasn't what I had in mind for GSoC so having to submit patches for every organization I am interested in kind of caught me off guard. I'll try to solve this quickly as this is the only thing missing from my proposal. :)

**Comment by GusAndrianos** [comment]
@v0dro That's awesome, I haven't really found anything that fits me better than SciRuby.

**Comment by GusAndrianos** [comment]
I am a bit confused. My test should check if a block of code can be passed and perform some action on a row already imported right?
How do I pass this row into the `from.csv` method? 

Can you give a usage example? I'm stuck on this for a while now.

**Comment by parthm** [comment]
Can we close this (based on discussion on #413)? #428 has been filed removal of block support.
